sky/resources.py

```diff
@@ -939,10 +935,17 @@ def less_demanding_than(
         if self.disk_tier is not None:
             if other.disk_tier is None:
                 return False
-            if self.disk_tier != other.disk_tier:
-                types = ['low', 'medium', 'high']
-                return types.index(self.disk_tier) < types.index(
-                    other.disk_tier)
+            if other.disk_tier == resources_utils.DiskTier.BEST:
+                return True
+            # Here, BEST tier means the best we can get; for a launched
+            # cluster, the best (and only) tier we can get is the launched
+            # cluster's tier.
+            if self.disk_tier == resources_utils.DiskTier.BEST:
+                return True
+            # resources_utils.DiskTier is listed in the order of increasing
+            # disk performance
+            types = list(resources_utils.DiskTier)
+            return types.index(self.disk_tier) < types.index(other.disk_tier)
```

Comment:
It would be nice to add some unit tests in https://github.com/skypilot-org/skypilot/blob/095d71fdcf66cb03131d7d3a4257e9d22dcdecd3/tests/test_jobs.py#L10

Reply:
Done. Thanks!